support/ebpf/sched_monitor.ebpf.c

```diff
@@ -33,3 +36,17 @@ int tracepoint__sched_process_free(struct sched_process_free_ctx *ctx)
 exit:
   return 0;
 }
+
+// tracepoint__sched_process_free is a tracepoint attached to the scheduler that frees processes.
+// Every time a processes exits this hook is triggered.
+SEC("tracepoint/sched/sched_process_free")
```

Comment:
We should avoid putting two different bpf programs into the same ELF section.

Reply:
AFAIK there's no issue with that, the eBPF ELF [specification](https://www.ietf.org/archive/id/draft-thaler-bpf-elf-00.html#name-text-sections) explicitly allows it. 

It's also a convenient way to group multiple programs under the same section with similar semantics. For example, we could have alternate versions of a tracepoint hook and the section name would tell us that they are all tracepoints and they all attach to the same point. Currently, we're not taking advantage of that but if we did, we could get rid of unnecessary code like the entire [AttachSchedMonitor](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/80155fde674817d74acf06659000820b46b72916/tracer/tracepoints.go#L32) method.

Reply:
I found another reference that mentions that depending on kernel version, the verifier could look at the entire section as a single program which could lead to failures depending on program logic. I'll split the functions into two sections with the same prefix so that we could take advantage of the information in the section name to automate loading/attaching if we wanted to in the future.